cirq/sim/density_matrix_simulator.py

```diff
@@ -528,7 +535,7 @@ def __init__(self, params: study.ParamResolver,
                          final_simulator_state=final_simulator_state)
         size = np.prod(protocols.qid_shape(self), dtype=int)
         self.final_density_matrix = np.reshape(
-            final_simulator_state.density_matrix, (size, size))
+            final_simulator_state.density_matrix, (size, size)).copy()
```

Comment:
IIUC, this state is only created once the simulation is complete, so it should be safe to not copy in this situation - no further mutation is possible.

Reply:
Yes but there are two ways to get this, one from the density_matrix on the last StepResult and also here.  One could manipulate the former and without this copy it could appear here on the final.  I think this would be odd because you'd have to opt into the no copy behavior for the StepResult, but I wanted to insure that never happens.

Reply:
Ahh, that makes sense. SGTM.